src/types/interfaces.ts

```diff
@@ -116,3 +116,12 @@ export interface DaiPermitMessage {
   expiry: BigNumberish
   allowed?: boolean
 }
+
+export interface NftMetadata {
+  name?: string
+  description?: string
+  image?: string
+  traits?: Array<object>
```

Comment:
Is the traits property needed? What for?

Reply:
Also, both EIPs 721 and 1155 tend to use a properties objects
`"properties": {
        "name": {
            "type": "string",
            "description": "Identifies the asset to which this token represents",
        }
    }`

Reply:
mm but on the EIPs they are defined like that: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md, https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1155.md#erc-1155-metadata-uri-json-schema